solana/src/balance_monitor.rs

```diff
@@ -0,0 +1,61 @@
+use crate::SolanaRpc;
+use solana_sdk::{signature::Keypair, signer::Signer};
+use std::{sync::Arc, time::Duration};
+
+// Check balance every 12 hours
+const DURATION: Duration = Duration::from_secs(43_200);
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("tokio join error")]
+    JoinError(#[from] tokio::task::JoinError),
+}
+
+pub async fn start(
+    app_account: &str,
+    solana: Option<Arc<SolanaRpc>>,
+    shutdown: triggered::Listener,
+) -> Result<futures::future::BoxFuture<'static, Result<(), Error>>, Error> {
+    Ok(match solana {
+        None => Box::pin(async move { Ok(()) }),
+        Some(rpc_client) => {
+            let app_metric_name = format!("{app_account}-sol-balance");
+            let handle =
+                tokio::spawn(async move { run(app_metric_name, rpc_client, shutdown).await });
+            Box::pin(async move {
+                match handle.await {
+                    Ok(()) => Ok(()),
+                    Err(err) => Err(Error::from(err)),
+                }
+            })
+        }
+    })
+}
+
+async fn run(metric_name: String, solana: Arc<SolanaRpc>, shutdown: triggered::Listener) {
```

Comment:
You should be able to make these generic over `SolanaNetwork` and not have to match on an Option

Reply:
ehhh I guess not